pkg/kncloudevents/message_receiver.go

```diff
@@ -45,6 +47,15 @@ func NewHTTPMessageReceiver(port int) *HTTPMessageReceiver {
 	}
 }
 
+// NewHTTPMessageReceiverWithChecker takes a handler func,
+// which runs as an additional health check in Drainer.
+func NewHTTPMessageReceiverWithChecker(port int, checker http.HandlerFunc) *HTTPMessageReceiver {
```

Comment:
Do you mind adding more explanation in the comment to when would one need a checker? maybe an example of a HTTPMessageReceiver with a health check? 

This is a pkg function, and the checker feature is not used in any of the eventing repo components (IMC, etc) so it's hard to understand how to use it unless one goes to a downstream repo that uses it (which I presume is the case here?)

Reply:
It is used in `knative-gcp`.

Reply:
Thanks for pointing it out, I will add release note and more explanation of the usage.
As for the use case, knative-gcp broker is using kn HTTPMessageReceiver, while needing drainer functionality, it is planned to add an authentication check in the form of health check. Previously, Drainer directly writes statusOK to kube probe request, and there is no place to add it.